General Base Program

0Arian opened the pull request:
**THIS IS USING PYTHON3**

## What is this?

This is the base generate_report.py program. It will take in a argument which is a pdf file and add a bar graph into your pdf of you're google analytics user session count.

## How do I run this?

You need several tools so run the following and make sure you have pip installed(pip3 if your pip is running python 2)

```
sudo pip install --upgrade google-api-python-client    
sudo pip install cairosvg    
sudo pip install pygal    
sudo pip install click
```

Next make sure you have these environment variables set

``GOOGLE_SERVICE_EMAIL``
``KEY_FILE_LOCATION``
This is your google service account email and the location to it's key file for API access, to create one simply head on over to
https://console.developers.google.com/permissions/serviceaccounts
Click Create service account.
In the Create service account window, type a name for the service account, and select Furnish a new private key.  When prompted for the key, select P12 and save it as any file name of your choice in any directory. Then list the location(including the key file in the location string) in the ``KEY_FILE_LOCATION`` environment variable. Also put in your newly created service account email which will be in this format
``<projectId>-<uniqueId>@developer.gserviceaccount.com``

After that is all done, simply run 
``python3 generate_report.py --file {{PDF-FILE-NAME-HERE}}``
or
``python3 generate_report.py``
Replace ``{{PDF-FILE-NAME-HERE}}`` with the filename of the pdf of your choice, if there no file argument then it will automatically insert an analytics.pdf in the current working directory


## Screenshot of example output
![ScreenShot](http://i.imgur.com/jiiTjgX.png)

0Arian committed:
Updated the gen report to what I originally had in the other repo

0Arian committed:
Made the variable names more professional and commented out all the test echo's

0Arian committed:
Fixed arguments for API Key

0Arian committed:
Updated generate report to print out 7, 14, 21, 28, and 35 days of activity

0Arian committed:
Fixed commenting changes and cleaned up get_results functions, then changed email and key file location to environment variables

0Arian committed:
Successful conversion from python2 into python3

0Arian committed:
type: feat

Imported PyGal and CairoSVG. With PyGal we are creating a bar chart and then rendering that chart into a svg file. With CairoSVG, we are taking the SVG file and pushing it into a pdf file.

0Arian committed:
type: feat - pdf arguments

Added argument parsing through click for PDF name.

davidawad commented on generate_report.py:
```diff
@@ -1,20 +1,139 @@
+"""
+---------------------------------------------------------------------------------
+Takes user sessions off Google Analytics API and Puts a bar graph into a PDF File
+Uses Python3
```
you don't need to include that it's written in python3 here

davidawad commented on generate_report.py:
```diff
@@ -1,20 +1,139 @@
+  xid21 = get_n_results(service, profile, 21)
+  xid28 = get_n_results(service, profile, 28)
+  xid35 = get_n_results(service, profile, 35)
+  print(int(xid28.get('rows')[0][0]))
```
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), you can do this in a for loop.

davidawad commented on generate_report.py:
```diff
@@ -1,20 +1,139 @@
+  bar_chart.add('28 Days', int(xid28.get('rows')[0][0]))
+  bar_chart.add('35 Days', int(xid35.get('rows')[0][0]))
+  bar_chart.render_to_file('bar_chart_sessions.svg')
+  cairosvg.svg2pdf(url='bar_chart_sessions.svg', write_to=file)
```
same as above :)

davidawad reviewed (changes requested):
Can you attach the actual pdf that this generates? and what the csv looks like that generates that PDF?

0Arian committed:
Redo: generate_report.py

Re-did the whole file with a lot of different methods and comments
Setting up the methods to generate data from the api with or without the dimensions
Starting to work the code for jinja compatability

0Arian committed:
IMPLEMENTED data for html documents

- Received all necessary data from google analytics to be passed into the html document

0Arian committed:
Setup Jinja to render google analytics data

- Render.html will now have access to all the data pulled from google
analytics
- Setup file that will be the rendered jinja template output

0Arian committed:
Added render.html

This is the file I'm using jinja to render

0Arian committed:
Updated render with charts

-Added all charts from ChartJS needed

0Arian committed:
Final Project setup

- Finished render.html template adding calendar icon, a table, a lot of
web design structuring with a styles.css file
- Phantomjs script capture.js is added to convert out html template into
a PNG file
- requirements updated with what is currently needed

TODO: README.md

0Arian committed:
FIXED small error

- Mistakenly pasted something in the wrong function, this is a fix

0Arian committed:
Bug fixed web template

- Chart widths fixed from being too wide
